pkg/logs/decoder/decoder.go

```diff
@@ -28,32 +28,49 @@ func NewInput(content []byte) *Input {
 	}
 }
 
-// Output represents a structured line.
-type Output struct {
+// DecodedInput represents a decoded line and the raw length
+type DecodedInput struct {
+	content    []byte
+	rawDataLen int
+}
+
+// NewDecodedInput returns a new decoded input.
+func NewDecodedInput(content []byte, rawDataLen int) *DecodedInput {
+	return &DecodedInput{
+		content:    content,
+		rawDataLen: rawDataLen,
+	}
+}
+
+// Message represents a structured line.
+type Message struct {
 	Content    []byte
 	Status     string
 	RawDataLen int
 	Timestamp  string
 }
 
-// NewOutput returns a new output.
-func NewOutput(content []byte, status string, rawDataLen int, timestamp string) *Output {
-	return &Output{
+// NewMessage returns a new output.
+func NewMessage(content []byte, status string, rawDataLen int, timestamp string) *Message {
+	return &Message{
 		Content:    content,
 		Status:     status,
 		RawDataLen: rawDataLen,
 		Timestamp:  timestamp,
 	}
 }
 
-// Decoder splits raw data into lines and passes them to a lineHandler that emits outputs
+// Decoder splits raw data into lines and passes them to a lineParser that passes them to
+// a lineHandler that emits outputs
+// Input->[decoder]->[parser]->[handler]->Output
```

Comment:
nits: If I understand the code correctly
```suggestion
// Input->[decoder]->[parser]->[handler]->Message
```